src/pages/index.js

```diff
@@ -118,65 +172,133 @@ function closeModal(modal) {
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitEditBtn.textContent = "Save";
+    });
 }
 
 function handleCardFormSubmit(evt) {
   evt.preventDefault();
-  const inputValue = { name: cardCaptionInput.value, link: cardUrlInput.value };
-  const cardEl = getCardElement(inputValue);
-  cardsList.prepend(cardEl);
-  closeModal(cardModal);
-  disableButton(cardSubmitButton, settings);
-  evt.target.reset();
+  const submitCardBtn = evt.submitter;
+  submitCardBtn.textContent = "Saving...";
+  api
+    .addNewCard({
+      name: cardCaptionInput.value,
+      link: cardUrlInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardsList.prepend(cardEl);
+      closeModal(cardModal);
+      disableButton(cardSubmitButton, settings);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitCardBtn.textContent = "Save";
+    });
 }
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving....";
+  api
+    .editAvatar({
+      avatar: avatarUrlInput.value,
+    })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarEditModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleModalClose(event) {
+  if (event.key === "Escape" && event.type === "keydown") {
+    const openedModalPopUp = document.querySelector(".modal_opened");
+    closeModal(openedModalPopUp);
+  }
+  if (event.target.classList.contains("modal")) {
+    closeModal(event.target);
+  }
+}
+
+//Listeners
 profileButtonEdit.addEventListener("click", () => {
-  const formElement = document.querySelector("#edit-modal .modal__form");
-  const inputList = formElement.querySelectorAll(".modal__input");
+  //editFormElement
+  //const formElement = document.querySelector("#edit-modal .modal__form");
+  const inputList = editFormElement.querySelectorAll(".modal__input");
   editModalDescriptionInput.value = profileDescription.textContent;
   editModalNameInput.value = profileName.textContent;
   openModal(editModal);
-  resetValidation(formElement, inputList, config);
+  resetValidation(editFormElement, inputList, config);
 });
 
-editModalCloseBtn.addEventListener("click", () => {
+//close button listeners
+/*editModalCloseBtn.addEventListener("click", () => {
   closeModal(editModal);
 });
 
-cardModalButton.addEventListener("click", () => {
-  openModal(cardModal);
+cardDeleteEscapebtn.addEventListener("click", () => {
+  closeModal(cardDeleteModal);
+});
+
+avatarCloseButton.addEventListener("click", () => {
+  closeModal(avatarEditModal);
 });
 
 closeModalButton.addEventListener("click", () => {
   closeModal(cardModal);
 });
-
+cancelButton.addEventListener("click", () => {
+  closeModal(cardDeleteModal);
```

Comment:
Please, return the `cancelButton` handling because the confirmation modal doesn't close when I click it


But I'll not reject only because if this mistake. You'll fix it easily

Reply:
Thank for extending graciously thank you. This has been fixed.